mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -2111,6 +2125,36 @@ def update(self, data, **kwargs) -> 'Block':
                 check_upstream_block_order=check_upstream_block_order,
             )
 
+        if 'downstream_blocks' in data and self.pipeline:
+            arr = (data.get('downstream_blocks') or [])
+            block_uuids_to_remove = \
+                [uuid for uuid in self.downstream_block_uuids if uuid not in arr]
+
+            for block_uuid in block_uuids_to_remove:
+                block = self.pipeline.get_block(block_uuid)
+                if not block:
+                    continue
+                block.update(
+                    dict(
+                        upstream_blocks=list(filter(
+                            lambda x, uuid=self.uuid: x != uuid,
+                            block.upstream_block_uuids or [],
+                        )),
+                    ),
+                    check_upstream_block_order=check_upstream_block_order,
+                )
+
+            for block_uuid in arr:
```

Comment:
Not all the block uuid in the `arr` are all new downstream block.

Reply:
I don’t understand what you’re suggesting here.